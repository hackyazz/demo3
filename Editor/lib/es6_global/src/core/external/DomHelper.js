

import * as StringService$WonderEditor from "../../service/atom/StringService.js";

function getBody (param){
  return document.body
};

function getAttribute (dom,prop){
  return dom.getAttribute(prop);
};

function locationReload (param){
  window.location.reload(true);
};

var apply = (
    function(dataArray, func) {
      return func.apply(null, dataArray);
    }
  );

var deleteKeyInMap = (
    function (key,map) {
      delete map[key];
      return map;
    }
  );

var getDomClientRect = (
  function(ele){
    if (!ele.getClientRects().length) {
      return { top: 0, left: 0 };
    }

    let rect = ele.getBoundingClientRect();
    if (rect.width || rect.height) {
        let doc = ele.ownerDocument;
        let win = doc.defaultView;
        let docElem = doc.documentElement;

        return {
            top: (rect.top + win.pageYOffset) - docElem.clientTop,
            left: (rect.left + win.pageXOffset) - docElem.clientLeft,
            offsetLeft: ele.offsetLeft,
            offsetTop: ele.offsetTop,
            height: rect.height,
            width: rect.width,
        }
    }

    return rect;
  }
  );

function getRandomKey(param) {
  return StringService$WonderEditor.floatToString(Date.now() * Math.random());
}

function intEl(n) {
  return String(n);
}

function textEl(str) {
  return str;
}

export {
  getBody ,
  getAttribute ,
  locationReload ,
  apply ,
  deleteKeyInMap ,
  getDomClientRect ,
  getRandomKey ,
  intEl ,
  textEl ,
  
}
/* apply Not a pure module */
