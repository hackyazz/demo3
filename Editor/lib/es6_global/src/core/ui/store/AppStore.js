

import * as Caml_exceptions from "../../../../../../node_modules/bs-platform/lib/es6/caml_exceptions.js";
import * as MapStore$WonderEditor from "./MapStore.js";
import * as UpdateStore$WonderEditor from "./UpdateStore.js";
import * as BottomShowComponentStore$WonderEditor from "../../composable_component/mainEditor/ui/store/BottomShowComponentStore.js";
import * as MainEditorInspectorStore$WonderEditor from "../../composable_component/mainEditor/composable_component/inspector/ui/store/MainEditorInspectorStore.js";
import * as ImmutableSparseMapService$WonderCommonlib from "../../../../../../node_modules/wonder-commonlib/lib/es6_global/src/ImmutableSparseMapService.js";

var ReplaceState = Caml_exceptions.create("AppStore-WonderEditor.ReplaceState");

var StartEngineAction = Caml_exceptions.create("AppStore-WonderEditor.StartEngineAction");

var MapAction = Caml_exceptions.create("AppStore-WonderEditor.MapAction");

var InspectorAction = Caml_exceptions.create("AppStore-WonderEditor.InspectorAction");

var UpdateAction = Caml_exceptions.create("AppStore-WonderEditor.UpdateAction");

var ShowComponentAction = Caml_exceptions.create("AppStore-WonderEditor.ShowComponentAction");

var state_001 = /* mapState : record */[/* componentsMap */undefined];

var state_002 = /* updateState : record */[/* componentTypeArr : array */[/* All */1]];

var state_003 = /* inspectorState : record */[/* showComponentMap */ImmutableSparseMapService$WonderCommonlib.createEmpty(/* () */0)];

var state_004 = /* showComponentState : record */[/* currentComponentType : Project */0];

var state = /* record */[
  /* isEditorAndEngineStart */false,
  state_001,
  state_002,
  state_003,
  state_004
];

function appReducter(state, action) {
  if (action === StartEngineAction) {
    return /* record */[
            /* isEditorAndEngineStart */true,
            /* mapState */state[/* mapState */1],
            /* updateState */state[/* updateState */2],
            /* inspectorState */state[/* inspectorState */3],
            /* showComponentState */state[/* showComponentState */4]
          ];
  } else if (action[0] === MapAction) {
    return /* record */[
            /* isEditorAndEngineStart */state[/* isEditorAndEngineStart */0],
            /* mapState */MapStore$WonderEditor.mapReducer(state[/* mapState */1], action[1]),
            /* updateState */state[/* updateState */2],
            /* inspectorState */state[/* inspectorState */3],
            /* showComponentState */state[/* showComponentState */4]
          ];
  } else if (action[0] === InspectorAction) {
    return /* record */[
            /* isEditorAndEngineStart */state[/* isEditorAndEngineStart */0],
            /* mapState */state[/* mapState */1],
            /* updateState */state[/* updateState */2],
            /* inspectorState */MainEditorInspectorStore$WonderEditor.inspectorReducer(state[/* inspectorState */3], action[1]),
            /* showComponentState */state[/* showComponentState */4]
          ];
  } else if (action[0] === UpdateAction) {
    return /* record */[
            /* isEditorAndEngineStart */state[/* isEditorAndEngineStart */0],
            /* mapState */state[/* mapState */1],
            /* updateState */UpdateStore$WonderEditor.updateReducer(state[/* updateState */2], action[1]),
            /* inspectorState */state[/* inspectorState */3],
            /* showComponentState */state[/* showComponentState */4]
          ];
  } else if (action[0] === ShowComponentAction) {
    return /* record */[
            /* isEditorAndEngineStart */state[/* isEditorAndEngineStart */0],
            /* mapState */state[/* mapState */1],
            /* updateState */state[/* updateState */2],
            /* inspectorState */state[/* inspectorState */3],
            /* showComponentState */BottomShowComponentStore$WonderEditor.showComponentReducer(state[/* showComponentState */4], action[1])
          ];
  } else if (action[0] === ReplaceState) {
    return action[1];
  } else {
    return state;
  }
}

export {
  ReplaceState ,
  StartEngineAction ,
  MapAction ,
  InspectorAction ,
  UpdateAction ,
  ShowComponentAction ,
  state ,
  appReducter ,
  
}
/* state Not a pure module */
