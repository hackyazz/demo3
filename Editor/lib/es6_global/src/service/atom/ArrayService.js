

import * as Curry from "../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Caml_obj from "../../../../../node_modules/bs-platform/lib/es6/caml_obj.js";
import * as Caml_option from "../../../../../node_modules/bs-platform/lib/es6/caml_option.js";
import * as Log$WonderLog from "../../../../../node_modules/wonder-log/lib/es6_global/src/Log.js";
import * as Contract$WonderLog from "../../../../../node_modules/wonder-log/lib/es6_global/src/Contract.js";
import * as Result$WonderEditor from "../../module/Result.js";
import * as OptionService$WonderEditor from "../primitive/OptionService.js";
import * as ArrayService$WonderCommonlib from "../../../../../node_modules/wonder-commonlib/lib/es6_global/src/ArrayService.js";
import * as StateEditorService$WonderEditor from "../state/editor/StateEditorService.js";
import * as MutableHashMapService$WonderCommonlib from "../../../../../node_modules/wonder-commonlib/lib/es6_global/src/MutableHashMapService.js";

function create(param) {
  return /* array */[];
}

function range(a, b) {
  var result = ArrayService$WonderCommonlib.createEmpty(/* () */0);
  for(var i = a; i <= b; ++i){
    result.push(i);
  }
  return result;
}

function unsafeGetFirst(arr) {
  return Contract$WonderLog.ensureCheck((function (r) {
                return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("first array element exist", "not"), (function (param) {
                              return Contract$WonderLog.assertNullableExist(r);
                            }));
              }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0), arr[0]);
}

function getFirst(arr) {
  return Caml_option.nullable_to_opt(arr[0]);
}

function unsafeGetLast(arr) {
  return Contract$WonderLog.ensureCheck((function (r) {
                return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("last array element exist", "not"), (function (param) {
                              return Contract$WonderLog.assertNullableExist(r);
                            }));
              }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0), arr[arr.length - 1 | 0]);
}

function getLast(arr) {
  return Caml_option.nullable_to_opt(arr[arr.length - 1 | 0]);
}

function removeLast(arr) {
  Contract$WonderLog.requireCheck((function (param) {
          return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("array length should >= 1", "not"), (function (param) {
                        return Contract$WonderLog.Operators[/* >= */7](arr.length, 1);
                      }));
        }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0));
  return OptionService$WonderEditor.unsafeGet(Caml_option.undefined_to_opt(arr.pop()));
}

function removeFirst(arr) {
  Contract$WonderLog.requireCheck((function (param) {
          return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("array length should >= 1", "not"), (function (param) {
                        return Contract$WonderLog.Operators[/* >= */7](arr.length, 1);
                      }));
        }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0));
  return OptionService$WonderEditor.unsafeGet(Caml_option.undefined_to_opt(arr.shift()));
}

function unsafeGetNth(index, arr) {
  return Contract$WonderLog.ensureCheck((function (r) {
                return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("array[" + (String(index) + "] element exist"), "not"), (function (param) {
                              return Contract$WonderLog.assertNullableExist(r);
                            }));
              }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0), arr[index]);
}

function hasItem(arr) {
  var match = arr.length > 0;
  if (match) {
    return true;
  } else {
    return false;
  }
}

function unshift(item, arr) {
  arr.unshift(item);
  return arr;
}

function push(item, arr) {
  arr.push(item);
  return arr;
}

function pushMany(itemArr, arr) {
  return ArrayService$WonderCommonlib.reduceOneParam((function (arr, item) {
                return push(item, arr);
              }), arr, itemArr);
}

function hasItemByFunc(func, arr) {
  return hasItem(arr.filter(Curry.__1(func)));
}

function removeDuplicateItems(buildKeyFunc, arr) {
  var resultArr = /* array */[];
  var map = MutableHashMapService$WonderCommonlib.createEmpty(/* () */0);
  for(var i = 0 ,i_finish = arr.length - 1 | 0; i <= i_finish; ++i){
    var item = arr[i];
    var key = buildKeyFunc(item);
    var match = MutableHashMapService$WonderCommonlib.get(key, map);
    if (match === undefined) {
      resultArr.push(item);
      MutableHashMapService$WonderCommonlib.set(key, item, map);
    }
    
  }
  return resultArr;
}

function exclude(excludeArr, arr) {
  return arr.filter((function (value) {
                return !excludeArr.includes(value);
              }));
}

function intersect(arr1, arr2) {
  return arr1.filter((function (value) {
                return arr2.includes(value);
              }));
}

function hasIntersect(arr1, arr2) {
  return intersect(arr1, arr2).length > 0;
}

function fastConcat(arr1, arr2) {
  return ArrayService$WonderCommonlib.reduceOneParam((function (arr1, value2) {
                return push(value2, arr1);
              }), arr1, arr2);
}

function fastConcatArrays(arrayArrays) {
  Contract$WonderLog.requireCheck((function (param) {
          return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("length >= 1", "not"), (function (param) {
                        return Contract$WonderLog.Operators[/* >= */7](arrayArrays.length, 1);
                      }));
        }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0));
  return ArrayService$WonderCommonlib.reduceOneParam(fastConcat, arrayArrays[0], arrayArrays.slice(1));
}

function isEqual(arr1, arr2) {
  return Caml_obj.caml_equal(arr1.sort(), arr2.sort());
}

function isInclude(sourceArr, targetArr) {
  return targetArr.filter((function (value) {
                return !sourceArr.includes(value);
              })).length === 0;
}

function _addFailureFunc(param, param$1) {
  return /* tuple */[
          param[0],
          Curry._1(param[1], param$1[1])
        ];
}

function _handleFailAndSuceessFunc(param, s) {
  return /* tuple */[
          param[0],
          Curry._1(param[1], s)
        ];
}

function _handleSuceessFuncAndFailFunc(func, param) {
  return /* tuple */[
          param[0],
          Curry._1(func, param[1])
        ];
}

function traverseSameDataResultAndCollectByApply(traverseFunc, arr) {
  var func = Result$WonderEditor.SameDataResult[/* apply */2];
  var applyFunc = function (param) {
    var func$1 = Curry._1(func, param);
    return (function (param) {
        return Curry._4(func$1, param, _addFailureFunc, _handleFailAndSuceessFunc, _handleSuceessFuncAndFailFunc);
      });
  };
  var returnFunc = Result$WonderEditor.SameDataResult[/* success */0];
  return ArrayService$WonderCommonlib.reduceOneParam((function (resultArr, value) {
                return applyFunc(applyFunc(Curry._1(returnFunc, push))(Curry._1(traverseFunc, value)))(resultArr);
              }), Curry._1(returnFunc, /* array */[]), arr);
}

export {
  create ,
  range ,
  unsafeGetFirst ,
  getFirst ,
  unsafeGetLast ,
  getLast ,
  removeLast ,
  removeFirst ,
  unsafeGetNth ,
  hasItem ,
  unshift ,
  push ,
  pushMany ,
  hasItemByFunc ,
  removeDuplicateItems ,
  exclude ,
  intersect ,
  hasIntersect ,
  fastConcat ,
  fastConcatArrays ,
  isEqual ,
  isInclude ,
  _addFailureFunc ,
  _handleFailAndSuceessFunc ,
  _handleSuceessFuncAndFailFunc ,
  traverseSameDataResultAndCollectByApply ,
  
}
/* Log-WonderLog Not a pure module */
