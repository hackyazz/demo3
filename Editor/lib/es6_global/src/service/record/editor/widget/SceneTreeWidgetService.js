


function getWidget(param) {
  return /* SceneTree */0;
}

function isWidget(startWidget) {
  if (startWidget !== undefined) {
    return startWidget === /* SceneTree */0;
  } else {
    return false;
  }
}

export {
  getWidget ,
  isWidget ,
  
}
/* No side effect */
