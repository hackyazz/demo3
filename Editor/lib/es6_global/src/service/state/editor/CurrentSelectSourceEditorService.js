


function getCurrentSelectSource(editorState) {
  return editorState[/* currentSelectSource */12];
}

function setCurrentSelectSource(currentSelectSource, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */currentSelectSource,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function clearCurrentSelectSource(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */undefined,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getCurrentSelectSource ,
  setCurrentSelectSource ,
  clearCurrentSelectSource ,
  
}
/* No side effect */
