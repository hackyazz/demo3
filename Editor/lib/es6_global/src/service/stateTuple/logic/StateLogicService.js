

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as StateEditorService$WonderEditor from "../../state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../state/engine/state/StateEngineService.js";
import * as DirectorEngineService$WonderEditor from "../../state/engine/DirectorEngineService.js";

function getEngineStateToGetData(handleFunc) {
  return Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0));
}

function getAndSetEngineState(handleFunc) {
  StateEngineService$WonderEditor.setState(Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function refreshEngineState(engineState) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, engineState));
  return /* () */0;
}

function refreshEngineStateAndReturnEngineState(engineState) {
  return DirectorEngineService$WonderEditor.loopBody(0, engineState);
}

function getAndRefreshEngineState(param) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function getAndRefreshEngineStateWithFunc(handleFunc) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0))));
  return /* () */0;
}

function loopBodyWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getAndRefreshEngineStateWhenStop(param) {
  StateEngineService$WonderEditor.setState(loopBodyWhenStop(StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function renderWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getEditorState(handleFunc) {
  return Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0));
}

function getAndSetEditorState(handleFunc) {
  StateEditorService$WonderEditor.setState(Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0)));
  return /* () */0;
}

function getStateToGetData(handleFunc) {
  return Curry._1(handleFunc, /* tuple */[
              StateEditorService$WonderEditor.getState(/* () */0),
              StateEngineService$WonderEditor.unsafeGetState(/* () */0)
            ]);
}

function getAndSetState(handleFunc) {
  var match = Curry._1(handleFunc, /* tuple */[
        StateEditorService$WonderEditor.getState(/* () */0),
        StateEngineService$WonderEditor.unsafeGetState(/* () */0)
      ]);
  StateEditorService$WonderEditor.setState(match[0]);
  StateEngineService$WonderEditor.setState(match[1]);
  return /* () */0;
}

function setState(param) {
  StateEditorService$WonderEditor.setState(param[0]);
  StateEngineService$WonderEditor.setState(param[1]);
  return /* () */0;
}

export {
  getEngineStateToGetData ,
  getAndSetEngineState ,
  refreshEngineState ,
  refreshEngineStateAndReturnEngineState ,
  getAndRefreshEngineState ,
  getAndRefreshEngineStateWithFunc ,
  loopBodyWhenStop ,
  getAndRefreshEngineStateWhenStop ,
  renderWhenStop ,
  getEditorState ,
  getAndSetEditorState ,
  getStateToGetData ,
  getAndSetState ,
  setState ,
  
}
/* StateEditorService-WonderEditor Not a pure module */
