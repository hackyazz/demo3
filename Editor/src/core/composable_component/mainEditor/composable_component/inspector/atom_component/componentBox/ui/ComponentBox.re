type state = {
  isShowComponent: bool,
  triangleDirection: string,
};

type action =
  | ToggleShowComponent;

module Method = {
  let removeComponent = AddableComponentRemoveComponentEventHandler.MakeEventHandler.pushUndoStackWithNoCopyEngineState;

  let changeShowComponentByType = ((uiState, dispatchFunc), type_, value) =>
    dispatchFunc(
      AppStore.InspectorAction(
        SetShowComponent(
          type_ |> InspectorComponentType.convertComponentTypeToInt,
          value,
        ),
      ),
    );
};

let component = ReasonReact.reducerComponent("ComponentBox");

let reducer = (reduxTuple, type_, action) =>
  switch (action) {
  | ToggleShowComponent => (
      state =>
        state.isShowComponent ?
          ReasonReactUtils.sideEffects(() =>
            Method.changeShowComponentByType(reduxTuple, type_, false)
          ) :
          ReasonReactUtils.sideEffects(() =>
            Method.changeShowComponentByType(reduxTuple, type_, true)
          )
    )
  };

let render =
    (
      reduxTuple,
      (header, isDisposable),
      (gameObject, gameObjectUIComponent, type_, title),
      {state, send}: ReasonReact.self('a, 'b, 'c),
    ) =>
  <article className="componentBox-component">
    <div className="header">
      <div
        className="header-triangle" onClick={_e => send(ToggleShowComponent)}>
        <span className={state.triangleDirection} />
      </div>
      <div
        className="header-title"
        title={
          switch (title) {
          | None => ""
          | Some(title) => title
          }
        }>
        {DomHelper.textEl(header)}
      </div>
      {
        isDisposable ?
          <div className="header-close">
            <img
              src="./public/img/close.png"
              onClick={
                _e => Method.removeComponent(reduxTuple, gameObject, type_)
              }
            />
          </div> :
          ReasonReact.null
      }
    </div>
    {state.isShowComponent ? gameObjectUIComponent : ReasonReact.null}
  </article>;

let make =
    (
      ~reduxTuple,
      ~header,
      ~isDisposable,
      ~gameObject,
      ~gameObjectUIComponent,
      ~isShowComponent,
      ~type_,
      ~title: option(string)=?,
      _children,
    ) => {
  ...component,
  initialState: () => {
    isShowComponent,
    triangleDirection: isShowComponent ? "triangle-bottom" : "triangle-right",
  },
  reducer: reducer(reduxTuple, type_),
  render: self =>
    render(
      reduxTuple,
      (header, isDisposable),
      (gameObject, gameObjectUIComponent, type_, title),
      self,
    ),
};