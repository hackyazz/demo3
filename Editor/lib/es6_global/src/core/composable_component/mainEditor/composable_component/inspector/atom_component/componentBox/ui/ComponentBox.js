

import * as Curry from "../../../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as React from "react";
import * as ReasonReact from "../../../../../../../../../../../node_modules/reason-react/lib/es6_global/src/ReasonReact.js";
import * as AppStore$WonderEditor from "../../../../../../../ui/store/AppStore.js";
import * as DomHelper$WonderEditor from "../../../../../../../external/DomHelper.js";
import * as ReasonReactUtils$WonderEditor from "../../../../../../../utils/ui/ReasonReactUtils.js";
import * as AddableComponentRemoveComponentEventHandler$WonderEditor from "../../addableComponent/ui/eventHandler/AddableComponentRemoveComponentEventHandler.js";

var removeComponent = AddableComponentRemoveComponentEventHandler$WonderEditor.MakeEventHandler[/* pushUndoStackWithNoCopyEngineState */0];

function changeShowComponentByType(param, type_, value) {
  return Curry._1(param[1], [
              AppStore$WonderEditor.InspectorAction,
              /* SetShowComponent */[
                type_,
                value
              ]
            ]);
}

var Method = /* module */[
  /* removeComponent */removeComponent,
  /* changeShowComponentByType */changeShowComponentByType
];

var component = ReasonReact.reducerComponent("ComponentBox");

function reducer(reduxTuple, type_, action) {
  return (function (state) {
      var match = state[/* isShowComponent */0];
      if (match) {
        return ReasonReactUtils$WonderEditor.sideEffects((function (param) {
                      return changeShowComponentByType(reduxTuple, type_, false);
                    }));
      } else {
        return ReasonReactUtils$WonderEditor.sideEffects((function (param) {
                      return changeShowComponentByType(reduxTuple, type_, true);
                    }));
      }
    });
}

function render(reduxTuple, param, param$1, param$2) {
  var send = param$2[/* send */3];
  var state = param$2[/* state */1];
  var title = param$1[3];
  var type_ = param$1[2];
  var gameObject = param$1[0];
  var match = state[/* isShowComponent */0];
  return React.createElement("article", {
              className: "componentBox-component"
            }, React.createElement("div", {
                  className: "header"
                }, React.createElement("div", {
                      className: "header-triangle",
                      onClick: (function (_e) {
                          return Curry._1(send, /* ToggleShowComponent */0);
                        })
                    }, React.createElement("span", {
                          className: state[/* triangleDirection */1]
                        })), React.createElement("div", {
                      className: "header-title",
                      title: title !== undefined ? title : ""
                    }, DomHelper$WonderEditor.textEl(param[0])), param[1] ? React.createElement("div", {
                        className: "header-close"
                      }, React.createElement("img", {
                            src: "./public/img/close.png",
                            onClick: (function (_e) {
                                return Curry._3(removeComponent, reduxTuple, gameObject, type_);
                              })
                          })) : null), match ? param$1[1] : null);
}

function make(reduxTuple, header, isDisposable, gameObject, gameObjectUIComponent, isShowComponent, type_, title, _children) {
  return /* record */[
          /* debugName */component[/* debugName */0],
          /* reactClassInternal */component[/* reactClassInternal */1],
          /* handedOffState */component[/* handedOffState */2],
          /* willReceiveProps */component[/* willReceiveProps */3],
          /* didMount */component[/* didMount */4],
          /* didUpdate */component[/* didUpdate */5],
          /* willUnmount */component[/* willUnmount */6],
          /* willUpdate */component[/* willUpdate */7],
          /* shouldUpdate */component[/* shouldUpdate */8],
          /* render */(function (self) {
              return render(reduxTuple, /* tuple */[
                          header,
                          isDisposable
                        ], /* tuple */[
                          gameObject,
                          gameObjectUIComponent,
                          type_,
                          title
                        ], self);
            }),
          /* initialState */(function (param) {
              return /* record */[
                      /* isShowComponent */isShowComponent,
                      /* triangleDirection */isShowComponent ? "triangle-bottom" : "triangle-right"
                    ];
            }),
          /* retainedProps */component[/* retainedProps */11],
          /* reducer */(function (param) {
              return reducer(reduxTuple, type_, param);
            }),
          /* jsElementWrapped */component[/* jsElementWrapped */13]
        ];
}

export {
  Method ,
  component ,
  reducer ,
  render ,
  make ,
  
}
/* component Not a pure module */
