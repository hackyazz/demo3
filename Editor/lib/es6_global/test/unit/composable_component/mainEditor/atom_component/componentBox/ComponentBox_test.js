

import * as List from "../../../../../../../../node_modules/bs-platform/lib/es6/list.js";
import * as Curry from "../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as React from "react";
import * as Sinon$1 from "sinon";
import * as ReasonReact from "../../../../../../../../node_modules/reason-react/lib/es6_global/src/ReasonReact.js";
import * as Wonder_jest from "../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as ReactTestRenderer from "react-test-renderer";
import * as AppStore$WonderEditor from "../../../../../../src/core/ui/store/AppStore.js";
import * as TestTool$WonderEditor from "../../../../../tool/TestTool.js";
import * as DomHelper$WonderEditor from "../../../../../../src/core/external/DomHelper.js";
import * as Caml_builtin_exceptions from "../../../../../../../../node_modules/bs-platform/lib/es6/caml_builtin_exceptions.js";
import * as ArrayService$WonderEditor from "../../../../../../src/service/atom/ArrayService.js";
import * as ComponentBox$WonderEditor from "../../../../../../src/core/composable_component/mainEditor/composable_component/inspector/atom_component/componentBox/ui/ComponentBox.js";
import * as ReactTestTool$WonderEditor from "../../../../../tool/ReactTestTool.js";
import * as GameObjectTool$WonderEditor from "../../../../../tool/GameObjectTool.js";
import * as StateLogicService$WonderEditor from "../../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../tool/MainEditorSceneTool.js";
import * as SceneTreeWidgetService$WonderEditor from "../../../../../../src/service/record/editor/widget/SceneTreeWidgetService.js";
import * as CurrentSelectSourceEditorService$WonderEditor from "../../../../../../src/service/state/editor/CurrentSelectSourceEditorService.js";

describe("componentBox", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        var _buildComponentBoxComponent = function (header, type_, isDisposable, gameObject) {
          return ReactTestRenderer.create(ReasonReact.element(undefined, undefined, ComponentBox$WonderEditor.make(/* tuple */[
                              TestTool$WonderEditor.buildEmptyAppState(/* () */0),
                              TestTool$WonderEditor.getDispatch(/* () */0)
                            ], header, isDisposable, gameObject, React.createElement("div", undefined, DomHelper$WonderEditor.textEl("simulate div component")), true, type_, /* array */[])));
        };
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                var partial_arg = SceneTreeWidgetService$WonderEditor.getWidget(/* () */0);
                return StateLogicService$WonderEditor.getAndSetEditorState((function (param) {
                              return CurrentSelectSourceEditorService$WonderEditor.setCurrentSelectSource(partial_arg, param);
                            }));
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("test component arguments", (function () {
                Wonder_jest.test("build componentBox component which can't be disposed should has no 'x'", (function () {
                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(_buildComponentBoxComponent("newCube", /* Transform */0, false, GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0)));
                      }));
                return Wonder_jest.test("build disposable componentBox component should has 'x'", (function () {
                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(_buildComponentBoxComponent("newCube", /* Transform */0, true, GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0)));
                            }));
              }));
        describe("test show component workflow", (function () {
                describe("test click triangle once to hide the common type component", (function () {
                        return Wonder_jest.test("test click close first box transform component, the component should be close", (function () {
                                      var dispatchFunc = Sinon.createEmptyStubWithJsObjSandbox(sandbox);
                                      ComponentBox$WonderEditor.reducer(/* tuple */[
                                              TestTool$WonderEditor.buildEmptyAppState(/* () */0),
                                              dispatchFunc
                                            ], /* Transform */0, /* ToggleShowComponent */0)(/* record */[
                                            /* isShowComponent */true,
                                            /* triangleDirection */""
                                          ]);
                                      var dispatchedAction = List.hd(Sinon.getArgs(Sinon.getCall(0, dispatchFunc)));
                                      if (dispatchedAction[0] === AppStore$WonderEditor.InspectorAction) {
                                        var action = dispatchedAction[1];
                                        return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](/* tuple */[
                                                        action[0],
                                                        action[1]
                                                      ]), /* tuple */[
                                                    /* Transform */0,
                                                    false
                                                  ]);
                                      } else {
                                        throw [
                                              Caml_builtin_exceptions.match_failure,
                                              /* tuple */[
                                                "ComponentBox_test.re",
                                                95,
                                                12
                                              ]
                                            ];
                                      }
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
