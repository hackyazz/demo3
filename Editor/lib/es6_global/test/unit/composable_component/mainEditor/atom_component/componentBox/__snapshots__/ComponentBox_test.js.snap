// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componentBox test component arguments build componentBox component which can't be disposed should has no 'x' 1`] = `
<article
  className="componentBox-component"
>
  <div
    className="header"
  >
    <div
      className="header-triangle"
      onClick={[Function]}
    >
      <span
        className="triangle-bottom"
      />
    </div>
    <div
      className="header-title"
    >
      newCube
    </div>
  </div>
  <div>
    simulate div component
  </div>
</article>
`;

exports[`componentBox test component arguments build disposable componentBox component should has 'x' 1`] = `
<article
  className="componentBox-component"
>
  <div
    className="header"
  >
    <div
      className="header-triangle"
      onClick={[Function]}
    >
      <span
        className="triangle-bottom"
      />
    </div>
    <div
      className="header-title"
    >
      newCube
    </div>
    <div
      className="header-close"
    >
      <img
        onClick={[Function]}
        src="./public/img/close.png"
      />
    </div>
  </div>
  <div>
    simulate div component
  </div>
</article>
`;
